account_chart_update/wizard/wizard_chart_update.py

```diff
@@ -620,16 +746,46 @@ def _find_fiscal_positions(self):
                 # Check the fiscal position for changes
                 fp = self.env['account.fiscal.position'].browse(fp_id)
                 notes = self.diff_notes(template, fp)
+
+                missing_xml_id = False
+                if self.recreate_xml_ids:
+                    missing_xml_id = self.missing_xml_id(fp)
+                    if missing_xml_id:
+                        notes += (notes and "\n" or "") + _("Missing XML-ID.")
+
                 if notes:
                     # Fiscal position template to be updated
                     wiz_fp.create({
                         'fiscal_position_id': template.id,
                         'update_chart_wizard_id': self.id,
                         'type': 'updated',
                         'update_fiscal_position_id': fp_id,
+                        'missing_xml_id': missing_xml_id,
                         'notes': notes,
                     })
 
+    def recreate_xml_id(self, template, model, res_id):
+        ir_model_data = self.env['ir.model.data']
+        template_xmlid = ir_model_data.search([
```

Comment:
If you already have the template, just use `template.get_external_id` instead

Reply:
get_external_id doesn't return the object of ir_model_data which is what I need, so to get it I have to write:

ext_id = template.get_external_id
and then
ext = self.env.ref ('ext_id')

There would be 2 connections to BD to find the object of ir_model_data.
What do you think?

Reply:
OK then